prompt/select.ts

```diff
@@ -99,6 +146,167 @@ export class Select<S extends SelectSettings = SelectSettings>
     return this.options[this.listIndex]?.value ?? this.settings.default;
   }
 
+  protected async submit(): Promise<void> {
+    const itemToSubmit = this.options[this.listIndex];
+
+    if (Select.itemIsBackButton(itemToSubmit)) {
+      const previousLevel = this.#parentCategories.pop();
+      if (typeof previousLevel === "object") {
+        this.options = previousLevel.options;
+        if (this.isCurrentlyInsideCategory()) {
+          this.listIndex = 2;
+        } else {
+          this.listIndex = 0;
+        }
+      }
+    } else if (Select.itemIsCategory(itemToSubmit)) {
+      this.#parentCategories.push({
+        options: this.options,
+        selectedCategoryIndex: this.listIndex,
+      });
+      this.options = this.itemsInsideCategory(itemToSubmit);
+      this.listIndex = 2;
+    } else {
+      await super.submit();
+    }
+  }
+
+  private itemsInsideCategory(
+    option: SelectOptionSettings,
+  ): SelectValueSettings {
+    return [
+      backItem,
+      ...Select.mapOptions([Select.separator(), ...option.options]),
+    ];
+  }
+
+  protected body(): string | Promise<string> {
+    let categoryText = "";
+
+    switch (this.#parentCategories.length) {
+      case 0:
+        return super.body();
+      case 1:
+        categoryText = dim(italic(this.nameOfParentCategory(0)));
+        break;
+
+      case 2: 
+        categoryText = dim(
+          italic(
+            this.nameOfParentCategory(0) + "/" + this.nameOfParentCategory(1),
+          ),
+        );
+        break;
+      
+      default: 
+        categoryText = dim(
+          italic(
+            this.nameOfParentCategory(0) + "/../" +
+              this.nameOfParentCategory(this.#parentCategories.length - 1),
+          ),
+        );
+        break;
+      
+    }
+
+    return categoryText + "\n" + super.body();
+  }
+
+  private nameOfParentCategory(index: number): string {
+    const { options, selectedCategoryIndex } = this.#parentCategories[index];
+    return options[selectedCategoryIndex].name;
+  }
+
+  protected match(): void {
+    const input: string = this.getCurrentInputValue().toLowerCase();
+
+    if (input === "") {
+      if (this.isCurrentlyInsideCategory()) {
+        const nearestParentCategories =
+          this.#parentCategories[this.#parentCategories.length - 1];
+        const categoryToRevertTo =
+          nearestParentCategories.options[nearestParentCategories.selectedCategoryIndex];
+
+        this.options = this.itemsInsideCategory(categoryToRevertTo);
+        this.listIndex = 2;
+      } else {
+        this.options = this.settings.options.slice();
+        this.clampListIndex();
+      }
+    } else {
+      const newOptions = this.itemsToSearchIn()
+        .filter((option) => Select.matchesItemOrSubItems(input, option));
+
+      this.options = this.buildSearchResult(input, newOptions);
```

Comment:
The way I make search results look good, is by taking the original options that were found, and then building a bunch of new options around those. This lets categories have their own items which are disabled, while their children appear enabled below.

Reply:
I think for now this is fine. But i think i would like to add support for nested options also to the checkbox prompt. To do this we would need to move some of your logic into the generic list class and than i think we don't need to flat all the options. But i would do this in a second PR.

Reply:
I'll stick with this then 😊 I guess in order to flatten it, navigation would need to not just work with the index of the selected item, but something more complex? Anyway, I'll let you see what you prefer 😄

Reply:
I have now moved the logic to the generic list class to add support for nested options for the checkbox prompt as well.
I also no longer disable the groups while searching because otherwise I won't be able to select all the items from that group while searching, and I think it's also a nice behavior if you can navigate through the groups while searching.